### PR TITLE
Support for symlinks in project references

### LINKED ISSUE #1114: Output while using project references is using the previous dependent package output instead of the current
Been trying to figure out how to set up a repo with yarn workspaces, webpack and typescript project references. Apologies if this is simply a configuration error on my part, but I'm unable to spot what I might have done wrong. Running on Windows 10

### Expected Behaviour
- Freshly clone https://github.com/jedster1111/ts-loader-project-references
- `yarn install`
- `yarn webpack` within the `a` workspace
  - I expect `b` to be built and write output files to `b/dist`
  - I also expect webpack to ouput `a/dist/index.js`
  - I can run `node index.js` inside of `a/dist` and see output `45`

- Change `b/src/index.ts` `getMeaningOfLife` to return `42`
- `yarn build` inside of `a`
  - I run `node index.js` inside of `a/dist` and see output `42`

### Actual Behaviour
The first `yarn build` will fail with:
```
yarn run v1.22.4
$ webpack
Hash: 32473a9cdaa03043ec8e
Version: webpack 4.43.0
Time: 808ms
Built at: 05/21/2020 1:01:35 AM
                       Asset       Size  Chunks             Chunk Names
     ../../b/dist/index.d.ts   53 bytes          [emitted]  
       ../../b/dist/index.js  149 bytes          [emitted]  
../../b/tsconfig.tsbuildinfo    6.3 KiB          [emitted]  
                  index.d.ts   11 bytes          [emitted]  
                    index.js   4.06 KiB    main  [emitted]  main
Entrypoint main = index.js
[./src/index.ts] 140 bytes {main} [built] [1 error]

ERROR in ./src/index.ts
Module not found: Error: Can't resolve 'b' in '/c/coding/jedster1111/ts-loader-project-references/a/src'
 @ ./src/index.ts 3:12-24

ERROR in /c/coding/jedster1111/ts-loader-project-references/a/src/index.ts
./src/index.ts
[tsl] ERROR in /c/coding/jedster1111/ts-loader-project-references/a/src/index.ts(1,32)
      TS2307: Cannot find module 'b' or its corresponding type declarations.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
If you check `b/dist` `index.js` does actually exist at this point.

If you run `yarn build` again, webpack will complete successfully.

Another interesting thing to note is if you update `getMeaningOfLife` to return `42`. Build webpack and run the built `index.js`. I get the old value, `45`. Re-run webpack and run the build file again, I now get `42`.

It seems like the dependant projects are getting built, but not actually used in the webpack output.

### Location of a Minimal Repository that Demonstrates the Issue.
https://github.com/jedster1111/ts-loader-project-references

### PR DESCRIPTION
**Pull request by sheetalkamat**
Fixes #1114

### CODE REVIEW

**Comment by johnnyreilly** [comment]
Nice work @sheetalkamat!

I can see there's a bunch of failing tests which you're no doubt looking into. Having had a quick look it seems that TypeScript <= 2.6 tests are failing.

I've been wondering about dropping support for older versions of TypeScript; maybe making the minimum supported version TypeScript 3.6. It would speed up our CI considerably as we'd have a much smaller test matrix to run in Travis / AppVeyor.  Users on older versions of TypeScript would still have older versions of ts-loader they could use.

If it makes your life easier, then do feel free to consider doing that in this PR. TypeScript 2.6 was a long time ago!

**Comment by johnnyreilly** [comment]
From the looks of it this is done @sheetalkamat?  Thanks so much for your work!  @andrewbranch would you be able to give this a review when you get a chance please?

**Comment by andrewbranch** [comment]
Yep, I will do it this week if not today. 🚀

File: test/comparison-tests/projectReferencesSymLinks/common/src/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const getMeaningOfLife2 = () => 45;
```
Comment (andrewbranch) [inline test/comparison-tests/projectReferencesSymLinks/common/src/index.ts:1]: 45?! 😱

Comment (johnnyreilly) [reply test/comparison-tests/projectReferencesSymLinks/common/src/index.ts:1]: That's a sick mind right there 😉

**Comment by johnnyreilly** [comment]
Shipping now https://github.com/TypeStrong/ts-loader/releases/tag/v8.0.0



Thanks for all your work @sheetalkamat and thanks for the review @andrewbranch!

**Pull request merged**